src/grid/simple.jl

```diff
@@ -52,7 +52,11 @@ create Arbitrary from grid.
         weight = zeros(Float64, size)
         for i in 1:size
             if i==1
-                weight[1] = 0.5*(grid[2]-grid[1])
+                if size==1
+                    weight[1] = 0 # allow arbitrary grid for 1 gridpoint, but integrate undefined
+                else
+                    weight[1] = 0.5*(grid[2]-grid[1])
```

Comment:
For better performance, it is better to exchange 58 and 56, because the former is much more common than the latter. In general, the second option requires an additional "jump" instruction, thus it is slower than the first option.